src/components/UploadButton.tsx

```diff
@@ -0,0 +1,37 @@
+import { ButtonHTMLAttributes, useRef } from "react";
+
+type UploadButtonProps = Omit<
+  ButtonHTMLAttributes<HTMLButtonElement>,
+  "onClick" | "type"
+> & {
+  onFile: (file: File) => void;
+  accept?: string;
+};
+
+export const UploadButton = ({
+  children,
+  onFile,
+  accept,
+  ...rest
+}: UploadButtonProps) => {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+
+  return (
+    <>
+      <input
+        accept={accept}
+        type="file"
+        onChange={({ target: { files } }) => files && onFile(files[0])}
+        style={{ display: "none" }}
```

Comment:
@GusBaamonde @jcamposmk This is better, right?

```suggestion
        hidden
```

Reply:
I think it is better than the inline style